### PR TITLE
Fixing tests

### PR DESCRIPTION
**Pull request by ahcorde**

### CODE REVIEW

**Review by vmayoral — APPROVED** [review]

**Comment by vmayoral** [comment]
Let's wait for the CI

**Review by vmayoral — CHANGES_REQUESTED** [review]
On a second look , this is actually not acceptable. You've changed the LOGGER names and that was agreed through a lot of iterations. Can you please revert this @ahcorde?

I mean, no changes to `LOGGER` variable names should be done.

File: moveit_core/background_processing/src/background_processing.cpp [hunk 1]
```diff
@@ -80,13 +80,13 @@ void BackgroundProcessing::processingThread()
       action_lock_.unlock();
       try
       {
-        RCLCPP_DEBUG(LOGGER, "Begin executing '%s'", action_name.c_str());
+        RCLCPP_DEBUG(LOGGER_BACKGROUND_PROCESSING, "Begin executing '%s'", action_name.c_str());
```
Comment (vmayoral) [inline moveit_core/background_processing/src/background_processing.cpp:83]: This shouldn't be reverted.

**Comment by ahcorde** [comment]
I can't otherwise it's not compiling

File: moveit_core/background_processing/src/background_processing.cpp [hunk 2]
```diff
@@ -42,7 +42,7 @@ namespace moveit
 namespace tools
 {
 // Logger
-rclcpp::Logger LOGGER = rclcpp::get_logger("background_processing");
+rclcpp::Logger LOGGER_BACKGROUND_PROCESSING = rclcpp::get_logger("background_processing");
```
Comment (vmayoral) [inline moveit_core/background_processing/src/background_processing.cpp:45]: This shouldn't be changed either.

File: moveit_core/background_processing/src/background_processing.cpp [hunk 3]
```diff
@@ -80,13 +80,13 @@ void BackgroundProcessing::processingThread()
...
-        RCLCPP_DEBUG(LOGGER, "Begin executing '%s'", action_name.c_str());
+        RCLCPP_DEBUG(LOGGER_BACKGROUND_PROCESSING, "Begin executing '%s'", action_name.c_str());
         fn();
-        RCLCPP_DEBUG(LOGGER, "Done executing '%s'", action_name.c_str());
+        RCLCPP_DEBUG(LOGGER_BACKGROUND_PROCESSING, "Done executing '%s'", action_name.c_str());
       }
       catch (std::exception& ex)
       {
-        RCLCPP_ERROR(LOGGER, "Exception caught while processing action '%s': %s", action_name.c_str(), ex.what());
+        RCLCPP_ERROR(LOGGER_BACKGROUND_PROCESSING, "Exception caught while processing action '%s': %s", action_name.c_str(), ex.what());
```
Comment (vmayoral) [inline moveit_core/background_processing/src/background_processing.cpp:89]: etc.

**Review by vmayoral — DISMISSED** [review]

**Comment by vmayoral** [comment]
> I can't otherwise it's not compiling

Let's look at this together.

**Comment by LanderU** [comment]
```bash
libmoveit_robot_model.a(robot_model.cpp.o): In function `moveit::core::RobotModel::constructShape(urdf::Geometry const*)':
robot_model.cpp:(.text+0x9116): undefined reference to `shapes::Sphere::Sphere(double)'
robot_model.cpp:(.text+0x918d): undefined reference to `shapes::Box::Box(double, double, double)'
robot_model.cpp:(.text+0x91d8): undefined reference to `shapes::Cylinder::Cylinder(double, double)'
robot_model.cpp:(.text+0x925d): undefined reference to `shapes::createMeshFromResource(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Eigen::Matrix<double, 3, 1, 0, 3, 1> const&)'
libmoveit_robot_model.a(link_model.cpp.o): In function `moveit::core::LinkModel::setGeometry(std::vector<std::shared_ptr<shapes::Shape const>, std::allocator<std::shared_ptr<shapes::Shape const> > > const&, std::vector<Eigen::Transform<double, 3, 1, 0>, Eigen::aligned_allocator<Eigen::Transform<double, 3, 1, 0> > > const&)':
link_model.cpp:(.text+0x720): undefined reference to `shapes::computeShapeExtents(shapes::Shape const*)'
link_model.cpp:(.text+0x77e): undefined reference to `typeinfo for shapes::Mesh'
link_model.cpp:(.text+0x785): undefined reference to `typeinfo for shapes::Shape'
collect2: error: ld returned 1 exit status
make[2]: *** [robot_model/test_moveit_robot_model] Error 1
```

**Comment by LanderU** [comment]
OS X:

```bash
CMake Error in robot_model/CMakeLists.txt:
  Imported target "resource_retriever::resource_retriever" includes
  non-existent path
    "/Users/osrf/jenkins-agent/workspace/packaging_osx/ws/install/include"
  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
  * The path was deleted, renamed, or moved to another location.
  * An install or uninstall procedure did not complete successfully.
  * The installation package was faulty and references files it does not
  provide.
```

**Comment by vmayoral** [comment]
@ahcorde, this is till failing. What's the status?

File: moveit_ros/planning/rdf_loader/src/rdf_loader.cpp [hunk 4]
```diff
@@ -53,23 +54,31 @@
...
+  qos.durability = RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL;
+
+  auto subscription_robot_description_ = node->create_subscription<std_msgs::msg::String>(
+    robot_description,
+    [&](std_msgs::msg::String::ConstSharedPtr msg) {
+      content = std::string(msg->data.c_str());
+  }, qos);
+
+  while(content.length()<1){
+    printf("Waiting for Robot model topic! Did you remap '%s'?\n", robot_description.c_str());
```
Comment (anasarrak) [inline moveit_ros/planning/rdf_loader/src/rdf_loader.cpp:79]: Please lets remove the printf and use RCLCPP_WARN or RCLCPP_INFO.

File: moveit_ros/planning/rdf_loader/src/rdf_loader.cpp [hunk 5]
```diff
@@ -83,11 +92,16 @@ rdf_loader::RDFLoader::RDFLoader(const std::string& robot_description)
...
-                    srdf_description.c_str());
-    return;
+  auto subscription_robot_description_semantic_ = node->create_subscription<std_msgs::msg::String>(
+    robot_description + "_semantic",
+    [&](std_msgs::msg::String::UniquePtr msg) {
+    scontent = std::string(msg->data.c_str());
+  }, qos);
+
+  while(scontent.length()<1){
+    printf("Waiting for Robot model semantic topic! Did you remap '%s'?\n", std::string(robot_description + "_semantic").c_str());
```
Comment (anasarrak) [inline moveit_ros/planning/rdf_loader/src/rdf_loader.cpp:102]: Same as https://github.com/AcutronicRobotics/moveit2/pull/56/commits/2cf435bf4dc98876a1253d3d5c53a28ebf8f15dd#r280375688

File: moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp [hunk 6]
```diff
@@ -47,11 +47,25 @@ RobotModelLoader::RobotModelLoader(const std::string& robot_description, bool lo
...
+{
+  Options opt(robot_description);
+  opt.load_kinematics_solvers_ = load_kinematics_solvers;
+  this->node_ = node;
   configure(opt);
 }
 
 RobotModelLoader::RobotModelLoader(const Options& opt)
 {
+  printf("RobotModelLoader Node is not declare!!\n");
```
Comment (anasarrak) [inline moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp:63]: Same as https://github.com/AcutronicRobotics/moveit2/pull/56/commits/2cf435bf4dc98876a1253d3d5c53a28ebf8f15dd#r280375688

File: moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp [hunk 7]
```diff
@@ -47,11 +47,25 @@ RobotModelLoader::RobotModelLoader(const std::string& robot_description, bool lo
 {
   Options opt(robot_description);
   opt.load_kinematics_solvers_ = load_kinematics_solvers;
+  printf("RobotModelLoader Node is not declare!!\n");
```
Comment (anasarrak) [inline moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp:50]: same https://github.com/AcutronicRobotics/moveit2/pull/56/commits/2cf435bf4dc98876a1253d3d5c53a28ebf8f15dd#r280375688

File: moveit_core/robot_state/src/robot_state.cpp [hunk 8]
```diff
@@ -1051,7 +1051,7 @@ void RobotState::getRobotMarkers(visualization_msgs::msg::MarkerArray& arr, cons
                                  bool include_attached) const
 {
   // ros::Time tm = ros::Time::now();
-
+  rclcpp::Clock clock_ros;
   builtin_interfaces::msg::Time tm = clock_ros.now();
```
Comment (anasarrak) [inline moveit_core/robot_state/src/robot_state.cpp:1055]:
```suggestion
   builtin_interfaces::msg::Time tm = rclcpp::Clock().now();
```
What about this change?

**Comment by ahcorde** [comment]
CI is configure with Crystal patch4, but we are using dashing (beta). We need to fix CI

**Comment by vmayoral** [comment]
@ahcorde, shall we close this one?

**Comment by ahcorde** [comment]
sure

**Pull request closed without merging**